substreams/ethereum-nabla/src/modules/map_components.rs

```diff
@@ -0,0 +1,195 @@
+use crate::abi;
+use anyhow::Result;
+use itertools::Itertools;
+
+use substreams::hex;
+use substreams_ethereum::pb::eth::v2::{Log, TransactionTrace};
+use substreams_ethereum::{pb::eth, Event};
+use substreams_helper::hex::Hexable;
+use tycho_substreams::prelude::{
+    Attribute, BlockTransactionProtocolComponents, ChangeType, FinancialType, ImplementationType,
+    ProtocolComponent, ProtocolType, TransactionProtocolComponents,
+};
+
+struct Portal {
+    contract_address: [u8; 20],
+    tx_hash: [u8; 32],
+}
+
+const NABLA_PORTAL: Portal = Portal {
+    contract_address: hex!("cB94Eee869a2041F3B44da423F78134aFb6b676B"),
+    tx_hash: hex!("7e9c5f39e41080aa7ab891ddd8669efc7191bae5906f74c2d5c7e7c12e219046"),
+};
```

Comment:
These are hardcoded contract address and transaction hash. The `tx_hash` provides a simple approach to prevent having to keep checking for deployment transaction after initial detection, which is senseless. While the protocol component for portal can be stored (in a `store`), state cannot be stored outside of the function using some "global" (or rather non-local) state (i.e. flag), since substreams doesn't operate like that (and the `store` only occurs after this first `map`, that is to say not available here at this time)